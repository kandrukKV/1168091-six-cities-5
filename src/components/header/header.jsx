import React from "react";
import {connect} from "react-redux";
import {NavLink} from "react-router-dom";
import PropTypes from "prop-types";
import {getAuthInfoSelector} from "../../store/selectors";
import {AppRoute} from "../../const";

const Header = ({authInfo}) => {
  return (
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">
            <NavLink to={AppRoute.ROOT} className="header__logo-link header__logo-link--active">
              <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81" height="41"/>
            </NavLink>
          </div>
          <nav className="header__nav">
            <ul className="header__nav-list">
              <li className="header__nav-item user">
                <NavLink to={AppRoute.FAVORITES} className="header__nav-link header__nav-link--profile">
                  <div className="header__avatar-wrapper user__avatar-wrapper">
                  </div>
                  <span className="header__user-name user__name">{authInfo ? authInfo.email : `Sign in`}</span>
                </NavLink>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
  );
};

Header.propTypes = {
  authInfo: PropTypes.oneOfType([
    PropTypes.oneOf([null]).isRequired,
    PropTypes.shape({
      email: PropTypes.string.isRequired
    }).isRequired
  ])
};

const mapStateToProps = (state) => ({
  authInfo: getAuthInfoSelector(state)
});

export {Header};
export default connect(mapStateToProps)(Header);
