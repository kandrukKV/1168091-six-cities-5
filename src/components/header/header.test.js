import React from "react";
import renderer from "react-test-renderer";
import {Header} from "./header";
import {BrowserRouter as Router} from 'react-router-dom';

const authInfo = {
  email: `email@email.com`
};

describe(`Header is rendered correctly`, () => {
  it(`Header rendered with auth info`, () => {
    const tree = renderer.create(
        <Router>
          <Header
            authInfo={authInfo}
          />
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Header rendered with out auth info`, () => {
    const tree = renderer.create(
        <Router>
          <Header
            authInfo={null}
          />
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});


