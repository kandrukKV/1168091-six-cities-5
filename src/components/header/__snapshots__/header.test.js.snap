// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header is rendered correctly Header rendered with auth info 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          aria-current="page"
          className="header__logo-link header__logo-link--active active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              aria-current={null}
              className="header__nav-link header__nav-link--profile"
              href="/favorites"
              onClick={[Function]}
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
              />
              <span
                className="header__user-name user__name"
              >
                email@email.com
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Header is rendered correctly Header rendered with out auth info 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          aria-current="page"
          className="header__logo-link header__logo-link--active active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              aria-current={null}
              className="header__nav-link header__nav-link--profile"
              href="/favorites"
              onClick={[Function]}
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
              />
              <span
                className="header__user-name user__name"
              >
                Sign in
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
